generic/grafana/setup/main.tf

```
@@ -0,0 +1,57 @@
+resource "grafana_contact_point" "this" {
+  name = var.contact_point_name
+
+  dynamic "slack" {
+    for_each = var.slack_config != null ? [1] : []
```

I would use the `count` instead of `for_each` for better readability:

```suggestion
    count = var.slack_config != null ? 1 : 0
```

---

When you are using a dynamic block, you can't use `count` you have to use `for_each`.